### PR TITLE
Update to cw 0.16

### LINKED ISSUE #75: Update template to 0.16.0
Seems like a lot of bumps are needed (see #63)

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #75.

### CODE REVIEW

**Review by ethanfrey — COMMENTED** [review]
Looks good, but no code changes needed?

Did you manually test generating a project and running tests (like the readme)?

I think some APIs changes to generate Response in 0.16, which is why I wonder

**Review by ethanfrey — APPROVED** [review]
Guess it really is that simple.

Can you update `Response::default()` to `Response::new()` btw? That is easier to extend. Maybe show an example emitting an attribute or event. (Those don't have to be in this PR before merge, but would be nice before we tag 0.16

**Comment by ethanfrey** [comment]
Oh, and can you bump to 0.8.0 (not -rc3) before merging?

File: Cargo.toml [hunk 1]
```diff
@@ -33,12 +33,12 @@ backtraces = ["cosmwasm-std/backtraces"]
...
 
 [dependencies]
-cosmwasm-std = { version = "0.15.2" }
-cosmwasm-storage = { version = "0.15.2" }
-cw-storage-plus = "0.7.0"
-schemars = "0.8.1"
-serde = { version = "1.0.125", default-features = false, features = ["derive"] }
+cosmwasm-std = { version = "0.16.0" }
+cosmwasm-storage = { version = "0.16.0" }
+cw-storage-plus = "0.8.0-rc3"
```
Comment (hashedone) [inline Cargo.toml:38]: 0.8.0 is already released, better to bump it to final version.

**Review by hashedone — APPROVED** [review]
Just bump last additional dependency, and then it is all ok. Don't forget to `cargo test` again to reupdate `Cargo.lock`.

File: src/contract.rs [hunk 2]
```diff
@@ -15,11 +15,15 @@ pub fn instantiate(
 ) -> Result<Response, ContractError> {
     let state = State {
         count: msg.count,
-        owner: info.sender,
+        owner: info.sender.clone(),
     };
     STATE.save(deps.storage, &state)?;
 
-    Ok(Response::default())
+    Ok(Response::new().add_attributes([
```
Comment (ethanfrey) [inline src/contract.rs:22]: Thanks for this demo

**Review by ethanfrey — APPROVED** [review]
Looking good

**Comment by ethanfrey** [comment]
Failed tests:

```
   Compiling testgen-ci v0.1.0 (/home/appveyor/projects/testgen-ci)
173error[E0277]: `[(&str, &str); 3]` is not an iterator
174  --> src/contract.rs:22:39
175   |
17622 |       Ok(Response::new().add_attributes([
177   |  _______________________________________^
17823 | |         ("method", "instantiate"),
17924 | |         ("owner", &info.sender.to_string()),
18025 | |         ("count", &msg.count.to_string()),
18126 | |     ]))
182   | |_____^ expected an implementor of trait `IntoIterator`
183   |
184   = note: the trait bound `[(&str, &str); 3]: IntoIterator` is not satisfied
185   = note: required because of the requirements on the impl of `IntoIterator` for `[(&str, &str); 3]`
186help: consider borrowing here
187   |
18822 |     Ok(Response::new().add_attributes(&[
18923 |         ("method", "instantiate"),
19024 |         ("owner", &info.sender.to_string()),
19125 |         ("count", &msg.count.to_string()),
19226 |     ]))
193   |
194
195error: aborting due to previous error
196
```

best would be to:

```
Ok(Response::new()
        .add_attribute("method", "instantiate")
        .add_attribute("owner", &info.sender)
        .add_attribute("count", msg.count.to_string())
)
```

`add_attributes` is usually for special cases, and only works if everything is the exact same type. I would show the chaining as a good example to follow.

**Review by orkunkl — APPROVED** [review]

**Comment by maurolacy** [comment]
> Failed tests:
> 
> ```
>    Compiling testgen-ci v0.1.0 (/home/appveyor/projects/testgen-ci)
> 173error[E0277]: `[(&str, &str); 3]` is not an iterator
> 174  --> src/contract.rs:22:39

Changed it. Funny that this worked locally just fine. Not sure why.

**Pull request merged**